### ISSUE TITLE
Artifact lock concurrency bug

### ISSUE THREAD

**Issue by originx** [gradle/gradle#2388]
We have recently started getting exceptions with locked artifact cache in our multi module builds.
Sadly we found no way to reproduce it so it just breaks our builds sporadically sometimes.
* What went wrong:
A problem occurred configuring project ':CommonCore'.
> Could not resolve all dependencies for configuration ':CommonCore:_debugPublish'.
   > Timeout waiting to lock artifact cache (/home/ci_de/.gradle/caches/modules-2). It is currently in use by another Gradle instance.
     Owner PID: 1272
     Our PID: 1416
     Owner Operation: 
     Our operation: 
     Lock file: /home/ci_de/.gradle/caches/modules-2/modules-2.lock

We can/are run/ning up to 5 gradle builds on a build machine in parallel.

### Expected Behavior
Locks for cache files works properly.
### Current Behavior
Sporadic failures for not being able to obtain locks on artifacts cache
### Context
We have jenkins multipipeline builds where we execute Android builds in paralel
### Steps to Reproduce (for bugs)
So far we found no way to reproduce this consistently, it just sometimes randomly breaks our CI:
We use a project similar to this:
https://github.com/originx/Gradle_troubles_android_enteprise

Ran this command from Jenkins pipeline on a project similar to the one above:

`./gradlew clean assemblePayGermanyCompatDebug compilePayGermanyCompatDebugUnitTestSources core:compileGermanyDebugUnitTestSources pay:compileGermanyDebugUnitTestSources compilePayGermanyCompatDebugAndroidTestSources core:compileGermanyDebugAndroidTestSources lintPayGermanyCompatDebug core:lintGermanyDebug pay:lintGermanyDebug testPayGermanyCompatDebugUnitTest testPayGermanyCompatDebugUnitTestCoverage core:testGermanyDebugUnitTest core:testGermanyDebugUnitTestCoverage pay:testGermanyDebugUnitTest pay:testGermanyDebugUnitTestCoverage spoonPayGermanyCompatDebugAndroidTest assemblePayGermanyCompatRelease compilePayGermanyCompatReleaseUnitTestSources core:compileGermanyReleaseUnitTestSources pay:compileGermanyReleaseUnitTestSources --profile --refresh-dependencies --stacktrace -PspoonDevice=emulator-5556`

### Your Environment
System:    Host: hostname Kernel: 3.10.0-514.6.1.el7.x86_64 x86_64 (64 bit) Console: tty 0
           Distro: CentOS Linux release 7.3.1611 (Core)
Machine:   Device: server System: Dell product: PowerEdge R330
Jenkins 2.52 
Jenkins pipeline plugin: 2.11
Gradle wrapper version: gradle-4.0
Android gradle plugin com.android.tools.build:gradle:2.3.3

 * Build scan URL: 

we cannot provide this since we are restricted by company policy from running such tools

### stacktraces
Here are the stacktraces:
https://gist.github.com/originx/8e1c5f2d0e6c245ae998d5c18227a90f

**Comment by bmuschko** [comment]
This is a duplicate of https://github.com/gradle/gradle/issues/851. I am going to close this issue. Feel free to reopen if you think there's more to be discussed.
